src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2186,5 +2186,182 @@ private async Task ValidateDeleteOrganizationAsync(Organization organization)
                 throw new BadRequestException("You cannot delete an Organization that is using Key Connector.");
             }
         }
+
+        public async Task DisableUserAsync(OrganizationUser organizationUser, Guid? disablingUserId)
+        {
+            if (organizationUser.Status == OrganizationUserStatusType.Disabled)
+            {
+                throw new BadRequestException("Already disabled.");
+            }
+
+            if (disablingUserId.HasValue && organizationUser.UserId == disablingUserId.Value)
+            {
+                throw new BadRequestException("You cannot disable yourself.");
+            }
+
+            if (organizationUser.Type == OrganizationUserType.Owner && disablingUserId.HasValue &&
```

Comment:
No need to check `disablingUserId.HasValue` again.

Reply:
This is a different check in combination with different criteria using `&&` and therefore is necessary since the prior `null` check is not exclusive as a singular criteria.